ChromecastSampleApp/app/src/main/java/com/ooyala/sample/common/PlayerActivity.java

```diff
@@ -0,0 +1,198 @@
+      pcode = lastChosenParams.getString("pcode", "");
+      domain = lastChosenParams.getString("domain", "");
+    }
+  }
+
+  @Override
+  protected void onRestart() {
+    super.onRestart();
+    Log.d(TAG, "Player Activity Restarted");
+    if (null != player) {
+      player.resume();
+    }
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    if (player != null) {
+      player.resume();
+    }
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    if (player != null) {
+      player.suspend();
+    }
+  }
+
+  protected void play(String ec) {
+    player.setEmbedCode(ec);
+    player.play();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    if (null != player) {
+      player.suspend();
+    }
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    if (requestCode == PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE) {
+      asked = true;
+      if (grantResults.length > 0 && grantResults[0] == PERMISSION_GRANTED) {
+        writePermission = true;
+      }
+    }
+  }
+
+  @Override
+  public void update(Observable arg0, Object argN) {
+    if (arg0 != player) {
+      return;
+    }
+    final String arg1 = OoyalaNotification.getNameOrUnknown(argN);
+    if (arg1 == OoyalaPlayer.TIME_CHANGED_NOTIFICATION_NAME) {
+      return;
+    }
+
+    if (arg1 == OoyalaPlayer.PLAY_COMPLETED_NOTIFICATION_NAME && secondEmbedCode != null) {
+      play(secondEmbedCode);
+      secondEmbedCode = null;
+    }
+
+    if (arg1.equalsIgnoreCase(OoyalaSkinLayoutController.FULLSCREEN_CHANGED_NOTIFICATION_NAME)) {
+      Log.d(TAG, "Fullscreen Notification received : " + arg1 + " - fullScreen: " + ((OoyalaNotification) argN).getData());
+    }
+
+
+    if (arg1 == OoyalaPlayer.ERROR_NOTIFICATION_NAME) {
+      final String msg = "Error Event Received";
+      if (player != null && player.getError() != null) {
+        Log.e(TAG, msg, player.getError());
+      } else {
+        Log.e(TAG, msg);
+      }
+    }
+
+    // Automation Hook: to write Notifications to a temporary file on the device/emulator
+    String text = getLog(argN);
+    playbackLog.writeToSdcardLog(text);
+    Log.d(TAG, "Notification Received: " + arg1 + " - state: " + player.getState());
+  }
+
+  protected String getLog(Object argN) {
+    final String arg1 = OoyalaNotification.getNameOrUnknown(argN);
+    return "Notification Received: " + arg1 + " - state: " + player.getState();
+  }
+
+  @Override
+  public void getTokenForEmbedCodes(List<String> embedCodes, EmbedTokenGeneratorCallback callback) {
+    String embedCodesString = "";
+    for (String ec : embedCodes) {
+      if (ec.equals("")) embedCodesString += ",";
+      embedCodesString += ec;
+    }
+
+    HashMap<String, String> params = new HashMap<>();
+    params.put("account_id", ACCOUNT_ID);
```

Comment:
Please add that commented code here : 
`    /* Uncommenting this will bypass all syndication rules on your asset
       This will not work unless you have a working API Key and Secret.
       This is one reason why you shouldn't keep the Secret in your app/source control */
//     params.put("override_syndication_group", "override_all_synd_groups");`

Reply:
done, thanks!